src/robo_snap.py

```diff
@@ -0,0 +1,198 @@
+"""ロボコンスナップを攻略するモジュール.
+
+@author: kawanoichi
+"""
+import os
+import time
+import numpy as np
+
+from detect_object import DetectObject, PROJECT_DIR_PATH
+from image_processing import ImageProcessing
+from official_interface import OfficialInterface
+
+
+class RoboSnap:
+    """ロボコンスナップ攻略クラス."""
+
+    img_list = ["FigA_1.png",
+                "FigA_2.png",
+                "FigA_3.png",
+                "FigA_4.png",
+                "FigB.png"]
+
+    fig_img_B = "FigB.png"
+
+    # NOTE:powershellの場合、絶対パスでbashファイルが実行できない？
+    #      よってcamera-system下での実装に対応
+    bash_path = "copy_fig.sh"
+    img_dir_path = os.path.join(PROJECT_DIR_PATH, "fig_image")
+
+    def __init__(self,
+                 raspike_ip="192.168.11.16",
+                 ) -> None:
+        """コンストラクタ.
+
+        Args:
+            raspike_ip: 走行体のIPアドレス
+        """
+        self.raspike_ip = raspike_ip
+        self.candidate_img_path = None
+
+    def execute_bash(self) -> None:
+        """bashファイルを実行する関数."""
+        for img in self.img_list:
+            bash_command = f"bash {self.bash_path} {self.raspike_ip} {img}"
+            try:
+                os.system(bash_command)
+            except Exception:
+                # 何回も実行するので、エラー文は省略
+                pass
+
+    def exist_image(self) -> str:
+        """画像が送られてきたかどうかの確認関数.
+
+        Returns:
+            img_name     (str): 画像ファイル名
+            img_path(str): 画像パス
+        """
+        for img_name in self.img_list:
+            img_path = os.path.join(self.img_dir_path, img_name)
+            if os.path.exists(img_path):
+                self.img_list.remove(img_name)
+                return img_name, img_path
+        return None, None
+
+    def check_bestshot(self, objects) -> int:
+        """ベストショット画像らしさスコアの算出.
+
+        信頼度は検出時にフィルターをかけているので検出したクラスと座標で判断する
+
+        Args:
+            objects: 検出した物体
+
+        Returns:
+            int: ベストショット画像らしさスコア
+
+        NOTE:
+            objectsの型について:
+                行数: 検出数
+                列数: 6列([x_min, y_min, x_max, y_max, conf, cls])
+
+            検出項目:
+                0: "Fig", 1: "FrontalFace", 2: "Profile"

```

Comment:
```suggestion
            検出項目:
                0: "Fig", 1: "FrontalFace", 2: "Profile"

```

上記の説明が欲しいです。Profileは横顔という意味ですか？